hack/install.sh

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o pipefail
+
+DEFAULT_BIN_DIR="/usr/local/bin"
+DEFAULT_CLI="karmadactl"
+CLI=${1:-"${DEFAULT_CLI}"}
+BIN_DIR=${2:-"${DEFAULT_BIN_DIR}"}
+GITHUB_REPO="karmada-io/karmada"
+
+# Helper functions for logs
+info() {
+    echo '[INFO] ' "$@"
+}
+
+warn() {
+    echo '[WARN] ' "$@" >&2
+}
+
+fatal() {
+    echo '[ERROR] ' "$@" >&2
+    exit 1
+}
+
+# Set os, fatal if operating system not supported
+setup_verify_os() {
+    if [[ -z "${OS}" ]]; then
+        OS=$(uname)
+    fi
+    case ${OS} in
+        Darwin)
+            OS=darwin
+            ;;
+        Linux)
+            OS=linux
+            ;;
+        *)
+            fatal "Unsupported operating system ${OS}"
+    esac
+}
+
+# Set arch, fatal if architecture not supported
+setup_verify_arch() {
+    if [[ -z "${ARCH}" ]]; then
+        ARCH=$(uname -m)
+    fi
+    case ${ARCH} in
+        arm64|aarch64|armv8l)
+            ARCH=arm64
+            ;;
+        amd64)
+            ARCH=amd64
+            ;;
+        x86_64)
+            ARCH=amd64
+            ;;
+        *)
+            fatal "Unsupported architecture ${ARCH}"
+    esac
+}
+
+# Set cli tools, install karmadactl by default, fatal if cli tool not supported
+setup_install_cli() {
+    case ${CLI} in
+        karmadactl)
+          CLI=karmadactl
+          ;;
+        kubectl-karmada)
+          CLI=kubectl-karmada
+          ;;
+        *)
+            fatal "Unsupported cli tool ${CLI}"
+    esac
+}
+
+# Verify existence of downloader executable
+verify_downloader() {
+    # Return failure if it doesn't exist or is no executable
+    [[ -x "$(which "$1")" ]] || return 1
+
+    # Set verified executable as our downloader program and return success
+    DOWNLOADER=$1
+    return 0
+}
+
+# Create tempory directory and cleanup when done
+setup_tmp() {
+    TMP_DIR=$(mktemp -d -t "${CLI}"-install.XXXXXXXXXX)
+    TMP_METADATA="${TMP_DIR}/${CLI}.json"
+    TMP_HASH="${TMP_DIR}/${CLI}.hash"
+    TMP_BIN="${TMP_DIR}/${CLI}.tgz"
+    cleanup() {
+        local code=$?
+        set +e
+        trap - EXIT
+        rm -rf "${TMP_DIR}"
+        exit ${code}
+    }
+    trap cleanup INT EXIT
+}
+
+# Find version from Github metadata
+get_release_version() {
+    if [[ -n "${KARMADA_VERSION}" ]]; then
```

Comment:
```suggestion
    if [[ -n "${KARMADA_VERSION-}" ]]; then
```

Or define this variable in the head this script, people can easily get all the variables they can use.